delphi/translators/for2py/genPGM.py

```diff
@@ -3474,18 +3498,52 @@ def create_grfn_dict(
     generator.fortran_file = original_file
 
     try:
-        filename_regex = re.compile(r"(.*)\..*$")
-        name_match = re.match(filename_regex, file_name)
-        assert name_match, f"Can't match filename to any format: {file_name}"
-        file_name = name_match.group(1)
+        filename_regex = re.compile(r"(?P<path>.*/)(?P<filename>.*).py")
+        file_match = re.match(filename_regex, file_name)
+        assert file_match, f"Can't match filename to any format: {file_name}"
+
+        path = file_match.group("path")
+        filename = file_match.group("filename")
+
+        # Since we do not have separate variable pickle file
+        # for m_*.py, we need to use the original program pickle
+        # file that module resides.
+        module_name = None
+        if module_file_exist:
+            module_file_path = file_name
+            module_name = filename[2:]  # First two elements are "m_"
+            org_file = get_original_file_name(original_file)
+            file_name = path + org_file
+        else:
+            file_name = path+filename
+            
         with open(f"{file_name}_variables_pickle", "rb") as f:
             variable_map = pickle.load(f)
         generator.variable_map = variable_map
     except IOError:
         raise For2PyError(f"Unable to read file {file_name}.")
 
+    # Extract variables with type that are declared in module
+    generator.module_variable_types = mode_mapper_dict[0]["symbol_types"]
+    # Extract functions (and subroutines) declared in module
+    for module in mode_mapper_dict[0]["subprograms"]:
+        for subp in mode_mapper_dict[0]["subprograms"][module]:
+            generator.module_subprograms.append(subp)
+
     grfn = generator.gen_grfn(asts, state, "")[0]
 
+    if len(generator.mode_mapper["use_mapping"]) > 0:
+        for user, module in generator.mode_mapper["use_mapping"].items():
+            if (
+                    (user in generator.module_names)
+                    or (module_name and module_name == user)
+            ):
+                module_paths = []
+                for import_mods in module:
+                    for mod_name, target in import_mods.items():
+                        module_path = path + "m_" + mod_name + "_GrFN.json"
```

Comment:
It looks like the assumption about "m_" prefix is everywhere.  Please look for them all carefully and fix them so that future changes to the naming scheme are easy to implement.

Reply:
@skdebray I've made the same change to `genPGM.py` like in `f2grfn.py`. All module file checking and forming will be done with `module_file_prefix` variable rather than hardcoded `"m_"` string.